build.sbt

```diff
@@ -1438,6 +1437,23 @@ lazy val `runtime-parser` =
     .dependsOn(syntax)
     .dependsOn(`syntax-rust-definition`)
 
+lazy val `runtime-compiler` =
+  (project in file("engine/runtime-compiler"))
+    .settings(
+      frgaalJavaCompilerSetting,
+      instrumentationSettings,
+      libraryDependencies ++= Seq(
+        "junit"          % "junit"           % junitVersion     % Test,
+        "com.github.sbt" % "junit-interface" % junitIfVersion   % Test,
+        "org.scalatest" %% "scalatest"       % scalatestVersion % Test,
+        "com.lihaoyi"   %% "fansi"           % fansiVersion
```

Comment:
There are two remaining dependencies we want to get rid in `runtime-compiler` (if we want to run it in VSCode extension as #7054 suggests or in the browser as proposed by #8094):
- we need to remove dependency on `fansi`
- we need to remove dependency on Truffle

Let those refactorings be part of some follow up PR.